### PR TITLE
Verify there is no more than one configuration

### PR DESCRIPTION
**Pull request by mshitrit**
#### Why we need this PR
SNR has a maximum of one agent created by it's daemonset on each node.
All of these agents are using a single configuration, therefore having more than one configuration may cause an unexpected behavior.
ATM there is no limitation on creating multiple SNR configuration, the purpose of this PR is to make sure there will be no more than one configuration for SNR.

#### Changes made
- Added a unique  annotation on the default configuration.
- Added a validation on creation of a configuration that this annotation exist, otherwise the create is rejected.

#### Which issue(s) this PR fixes
[ECOPROJECT-1997](https://issues.redhat.com//browse/ECOPROJECT-1997)

If there is not a correspondent issue yet, you might want to open a new one yourself, describing the problem you observed.
-->

### CODE REVIEW

File: api/v1alpha1/selfnoderemediationconfig_webhook.go [hunk 1]
```diff
@@ -172,3 +175,10 @@ func validateToleration(toleration v1.Toleration) error {
 	}
 	return nil
 }
+
+func (r *SelfNodeRemediationConfig) validateDefault() error {
+	if defaultAnnotationVal, isDefaultAnnotationExist := r.Annotations[utils.IsDefaultConfigurationAnnotation]; isDefaultAnnotationExist && defaultAnnotationVal == "true" {
+		return nil
+	}
+	return fmt.Errorf("only single configuration is allowed")
```
Comment (razo7) [inline api/v1alpha1/selfnoderemediationconfig_webhook.go:183]: NIT: maybe be more specific?
```suggestion
	return fmt.Errorf("only single SNRConfig is allowed in the cluster")
```

File: api/v1alpha1/selfnoderemediationconfig_webhook.go [hunk 2]
```diff
@@ -172,3 +175,10 @@ func validateToleration(toleration v1.Toleration) error {
 	}
 	return nil
 }
+
+func (r *SelfNodeRemediationConfig) validateDefault() error {
+	if defaultAnnotationVal, isDefaultAnnotationExist := r.Annotations[utils.IsDefaultConfigurationAnnotation]; isDefaultAnnotationExist && defaultAnnotationVal == "true" {
+		return nil
+	}
+	return fmt.Errorf("only single SelfNodeRemediationConfig is allowed in the cluster")
+}
```
[the remark this answers is not in the document]
Comment (mshitrit) [reply api/v1alpha1/selfnoderemediationconfig_webhook.go:184]: Very good points.
I agree documentation is indeed required.
Regarding the name how about using `IsSingeltonConfigurationAnnotation = "is-singelton-configuration.self-node-remediation.medik8s.io"` ?

[the remark this answers is not in the document]
Comment (clobrano) [reply api/v1alpha1/selfnoderemediationconfig_webhook.go:184]: It's surely better, however I still feel we're not really doing much with this change :sweat_smile: 
If we need documentation anyway, maybe just the documentation will be enough :shrug:

[the remark this answers is not in the document]
Comment (slintes) [reply api/v1alpha1/selfnoderemediationconfig_webhook.go:184]: And how do docs prevent creation of multiple configs with that annotation?
IMHO the only way to fix the issue is to use a hardcoded name for the config, and prevent creation of configs with any other name...

[the remark this answers is not in the document]
Comment (slintes) [reply api/v1alpha1/selfnoderemediationconfig_webhook.go:184]: suggestion: 
- use `cluster` (edit: nope, that will break existing configs, use what we used so far) as name, that's what OCP uses as well for cluster wide configs. Can be enforced with kubebuilder by making it an enum with a single value
- validating webhook just ensures that it's in the same namespace as the operator

[the remark this answers is not in the document]
Comment (mshitrit) [reply api/v1alpha1/selfnoderemediationconfig_webhook.go]: Nice idea of using a hard coded name + validate the ns :+1:

**Review by clobrano — CHANGES_REQUESTED** [review]
I left one minimal comment, otherwise lgtm

**Review by clobrano — APPROVED** [review]

**Comment by clobrano** [comment]
Giving a chance to get more reviews
/hold

**Review by slintes — APPROVED** [review]

**Pull request merged**